### PR TITLE
Fixing m3tsz int overflow edge case

### PR DESCRIPTION
**Pull request by martin-mao**
Fixing case where m3tsz int optimization overflows with very large ints.

### CODE REVIEW

**Comment by xichen2020** [comment]
I assume the failing test is unrelated to this change?

File: encoding/m3tsz/roundtrip_test.go [hunk 1]
```diff
@@ -220,3 +225,20 @@ func generateMixedDatapoints(numPoints int, timeUnit time.Duration) []ts.Datapoi
 	}
 	return res
 }
+
+func generateOverflowDatapoints() []ts.Datapoint {
+	var startTime int64 = 1427162462
+	currentTime := time.Unix(startTime, 0)
+	largeInt := float64(math.MaxInt64 - 1)
```
Comment (robskillington) [inline encoding/m3tsz/roundtrip_test.go]: Shouldn't this be `float64(math.MaxInt64)-1` to be more safe about compiler semantics here?  Same with the variable below.
